Make first installed candidate version automatically the default one

pbassiner opened the pull request:
This fixes #442 
The tests should be reviewed carefully. I had to specify `And the system is bootstrapped` on each test because I needed it to be executed after a `Given the candidate "grails" version "1.3.9" is already installed and default` (checked out the tests of the `default` command for that matter).

marc0der commented on src/test/cucumber/install_candidate.feature:
```diff
@@ -3,62 +3,64 @@ Feature: Install Candidate
     When I enter "sdk install grails 1.3.9"
     Then I see "Stop! grails 1.3.9 is already installed."
 
-  Scenario: Install a candidate and select to use it
```
Don't think this should be deleted, as it test a very valid case.

pbassiner replied:
If adapted to the new feature, I think it would be the same test as `Install a specific Candidate`. That's why I removed it.

marc0der replied:
It's subtly different because `Install a specific candidate` is the first time a version is installed (so no current default exists). I'm adding the scenario that tests this case too.

pbassiner commented:
Regarding your first comment, that's what I think this is doing:
* If there's no current version for a candidate, it sets the installed version as default
* Otherwise, it asks for confirmation on setting a new candidate version as default

The last two tests I added (`Install a candidate and select to use it automatically` and `Install a candidate and do not select to use it`) aim to cover both scenarios.

I've already rebased to master and updated the `-n "$CURRENT"` check.

Let me know if I'm missing something regarding the new scenarios and its related tests.

marc0der commented:
I've merged this locally so it doesn't seem to be showing up on github. Thanks for yet another great contribution! Keep them coming!
